galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/spi/gherkin/GherkinTest.java

```diff
@@ -0,0 +1,80 @@
+package dev.galasa.framework.spi.gherkin;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import dev.galasa.framework.TestRunException;
+import dev.galasa.framework.spi.IRun;
+
+public class GherkinTest {
+
+    private List<GherkinMethod> methods;
+    private URI gherkinUri;
+
+    private String testName;
+    private List<String> comments;
+
+    public GherkinTest(IRun run) throws TestRunException {
+        this.methods = new ArrayList<>();
+        this.comments = new ArrayList<>();
+
+        try {
+            gherkinUri = new URI(run.getGherkin());
+
+            if (gherkinUri.getScheme().equals("file")) {
+                File gherkinFile = new File(gherkinUri);
+                BufferedReader br = new BufferedReader(new FileReader(gherkinFile));
+
+                String line;
+                GherkinMethod currentMethod = null;
+                while ((line = br.readLine()) != null) {
+                    line = line.trim();
+                    if(line.isEmpty()) {
+                        continue;
+                    }
+
+                    if(line.startsWith("Feature:")) {
```

Comment:
Read the file in,   then parse with REGEX as this is rather simplistic at the moment compared to all the "statement" types we will need to support